apps/blockchain/lib/blockchain/interface/account_interface.ex

```diff
@@ -37,9 +40,144 @@ defmodule Blockchain.Interface.AccountInterface do
     }
   end
 
-  @spec commit_storage(t()) :: EVM.state()
+  @spec commit(t()) :: t()
+  def commit(account_interface) do
+    account_interface.cache
+    |> Cache.commit(account_interface.state)
+    |> new()
+  end
+
+  @spec commit_storage(t()) :: t()
   def commit_storage(account_interface) do
-    Cache.commit(account_interface.cache, account_interface.state)
+    new_state = Cache.commit_storage(account_interface.cache, account_interface.state)
+
+    new(new_state, %Cache{accounts_cache: account_interface.cache.accounts_cache})
+  end
+
+  @spec commit_accounts(t()) :: t()
+  def commit_accounts(account_interface) do
+    new_state = Cache.commit_accounts(account_interface.cache, account_interface.state)
+
+    new(new_state, %Cache{storage_cache: account_interface.cache.storage_cache})
+  end
+
+  @spec increment_account_nonce(t(), Address.t()) :: t()
+  def increment_account_nonce(account_interface, address) do
+    {account, code} = account(account_interface, address)
+    updated_account = %{account | nonce: account.nonce + 1}
+
+    updated_cache =
+      Cache.update_account(account_interface.cache, address, {updated_account, code})
+
+    %{account_interface | cache: updated_cache}
+  end
+
+  @spec transfer_wei!(t(), Address.t(), Address.t(), EVM.Wei.t()) :: t()
+  def transfer_wei!(account_interface, from, to, wei) do
+    {from_account, from_account_code} = account(account_interface, from)
+
+    cond do
+      wei < 0 ->
+        raise("wei transfer cannot be negative")
```

Comment:
Is there a reason to use `raise` over  returning an `:error/:ok` tuple here?

Reply:
this method is replacing `Account.transfer!` method that is used in AccountInterface protocol implementation. `Account.transfer!` also raises errors